Framework/Geometry/src/Instrument/ParameterMap.cpp

```diff
@@ -1244,13 +1236,29 @@ void ParameterMap::setInstrument(const Instrument *instrument) {
                            "base instrument, not a parametrized instrument");
   m_instrument = instrument;
 
+  if (empty() && instrument->componentInfo()) {
```

Comment:
This is good, and a very important mechanism. Maybe worth extracting to a named method?